### PR TITLE
rework calculator example

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: public/content/en/basics.md [hunk 1]
```diff
@@ -334,18 +334,18 @@ void main()
     // and END, whereas END is NOT inclusive.
     // Depending on the result we call one of
     // the math operations.
-    switch (uniform(0,4)) {
+    switch (uniform(0, 4)) {
         case 0:
-            writeln(add(a,b));
+            add(a, b).writeln;
```
Comment (stonemaster) [inline public/content/en/basics.md]: UFCS hasn't be introduced here

File: public/content/en/basics.md [hunk 2]
```diff
@@ -334,18 +334,18 @@ void main()
...
             break;
         case 1:
-            writeln(sub(a,b));
+            sub(a, b).writeln;
```
Comment (stonemaster) [inline public/content/en/basics.md]: ditto

File: public/content/en/basics.md [hunk 3]
```diff
@@ -334,18 +334,18 @@ void main()
...
             break;
         case 2:
-            writeln(mul(a,b));
+            mul(a, b).writeln;
```
Comment (stonemaster) [inline public/content/en/basics.md]: ditto

File: public/content/en/basics.md [hunk 4]
```diff
@@ -334,18 +334,18 @@ void main()
...
             break;
         case 3:
-            writeln(div(a,b));
+            div(a, b).writeln;
```
Comment (stonemaster) [inline public/content/en/basics.md]: ditto

File: public/content/en/basics.md [hunk 5]
```diff
@@ -354,11 +354,14 @@ void main()
...
-// NOTE:
-//   add(), sub(), mul() and div()
-//   are NOT visible outside of main!
-
+void main()
+{
+    randomCalculator();
+    //  add(), sub(), mul() and div()
+    //  are NOT visible outside of main!
+    static assert(!__traits(compiles, add(1, 2)));
```
Comment (stonemaster) [inline public/content/en/basics.md]: spurious new line

**Comment by wilzbach** [comment]
rebased.

File: public/content/en/basics.md [hunk 6]
```diff
@@ -354,11 +354,15 @@ void main()
...
-//   add(), sub(), mul() and div()
-//   are NOT visible outside of main!
-
+void main()
+{
+    randomCalculator();
+    //  add(), sub(), mul() and div()
+    //  are NOT visible outside of main!
+    static assert(!__traits(compiles,
+    						add(1, 2)));
```
Comment (stonemaster) [inline public/content/en/basics.md]: indentation

**Comment by wilzbach** [comment]
rebased

**Comment by stonemaster** [comment]
@wilzbach There are merge conflicts maybe because of another pull request

**Comment by wilzbach** [comment]
> @wilzbach There are merge conflicts maybe because of another pull request

Yep it's rebased now - it doesn't show much anymore except for proofing that the symbols aren't available anymore.

**Comment by wilzbach** [comment]
Note that #164 does this in a similar way - feel free to choice which you like more.

**Comment by wilzbach** [comment]
rebased.

**Comment by stonemaster** [comment]
Unfortunately there are merge conflicts. Otherwise perfect!

**Comment by wilzbach** [comment]
> Unfortunately there are merge conflicts. Otherwise perfect!

Turns out I already fixed that in another PR, but putting it in `main` sounds better to me.

File: public/content/en/basics.md [hunk 7]
```diff
@@ -404,11 +404,15 @@ void main()
...
-// are NOT visible outside of main!
-
-assert(!__traits(compiles, add(1, 2)));
+void main()
+{
+    randomCalculator();
+    //  add(), sub(), mul() and div()
+    //  are NOT visible outside of their scope
+    static assert(!__traits(compiles,
+                            add(1, 2)));
```
Comment (stonemaster) [inline public/content/en/basics.md]: Unneeded whitespace

File: public/content/en/basics.md [hunk 8]
```diff
@@ -404,11 +404,15 @@ void main()
...
 
+}
```
Comment (stonemaster) [inline public/content/en/basics.md]: Better separate } with empty line from next section

**Comment by wilzbach** [comment]
>Better separate } with empty line from next section

Sorry that was an artifact from the merge.

**Comment by stonemaster** [comment]
Reopen because compilation container was fixed.

**Comment by wilzbach** [comment]
Rebased after the file split - tests should be passing now.

**Pull request merged**